Validate accepted emendation and test

marc-va opened the pull request:
#### :tophat: What? Why?


#### :pushpin: Related Issues
- Related to #?
- Fixes #?

#### :clipboard: Subtasks
- [x] Add `CHANGELOG` entry
- [ ] Add documentation regarding the feature 
- [ ] Add/modify seeds
- [ ] Add tests
- [ ] Another subtask

### :camera: Screenshots (optional)
![Description](URL)

agustibr committed:
Amendment feature activated in the proposal component admin panel and render Ammend Button

agustibr committed:
Amendment feature activated in the proposal component admin panel and render Ammend Button

agustibr committed:
Amendment feature activated in the proposal component admin panel and render Ammend Button

agustibr committed:
Amendment feature activated in the proposal component admin panel and render Ammend Button

agustibr committed:
Amendment feature activated in the proposal component admin panel and render Ammend Button

agustibr committed:
Amendment model, Amendable concern, amendments_list, emendation_announcement, emendation_actions ...

agustibr committed:
Amendment model, Amendable concern, amendments_list, emendation_announcement, emendation_actions ...

agustibr committed:
Amendment model, Amendable concern, amendments_list, emendation_announcement, emendation_actions ...

agustibr committed:
Refactored due to review

agustibr committed:
Refactored due to review

agustibr committed:
Refactored due to review

agustibr committed:
Refactored due to review

agustibr committed:
Refactored due to review

agustibr committed:
Refactored due to review

agustibr committed:
Enable UserGroup to Amend

agustibr committed:
Enable UserGroup to Amend

agustibr committed:
Enable UserGroup to Amend

agustibr committed:
moved to amendable dir

fix docs

agustibr committed:
add @emendation_form to proposals_controller

agustibr committed:
add evaluating? method

agustibr committed:
Accept Amendment

agustibr committed:
add :emendation and :amendment to EmendationActionsCell

agustibr committed:
Add Amended by list in amendable detail

agustibr committed:
Fix Created notice

agustibr committed:
fix flash type on accepted error

agustibr committed:
fix flash type on accepted error

agustibr committed:
Use CollapsibleListCell for amendments list

agustibr committed:
add action_authorized_link_to to amend_button_card

agustibr committed:
Added Amend system Specs for Creation

agustibr committed:
Added Amend system Specs for Creation

agustibr committed:
Added Amend system Specs for Creation

agustibr committed:
configurable amendable fields

agustibr committed:
don't ignore proposal reference

agustibr committed:
Update amendable/create to perform_action

agustibr committed:
Permit and admin react on an official emmendation

agustibr committed:
Add "Amendment" label to CardM

agustibr committed:
add hashtags support

agustibr committed:
minor improvements for amendables

aitorlb committed:
Merge proposal/emendation state fixes

aitorlb committed:
Add accept amendment spec

aitorlb committed:
Add accept amendment system test

marc-va committed:
Accept emendation validate errors and test

agustibr commented on decidim-core/spec/forms/decidim/amendable/review_form_spec.rb:
```diff
@@ -0,0 +1,77 @@
+
+
+      let(:title) { "Loura Hansen II 1" }
+      let(:body) {"Dlksdjfklesjfklew lkdjflksdjflk sdlkfjsdlkfjskdjf lskdfj skjflk sjflksdjf lksjflksdjflks jflksd jlkdsjlckjksd"}
```
DummyResource has no `:body` attribute

https://github.com/CodiTramuntana/decidim/blob/2acf010429b0ead4722fca6db8f9e307a45152a9/decidim-dev/app/forms/decidim/dummy_resources/dummy_resource_form.rb

agustibr commented on decidim-core/app/forms/decidim/amendable/review_form.rb:
```diff
@@ -13,7 +13,7 @@ class ReviewForm < Decidim::Amendable::Form
       attribute :user_group_id, Integer
 
-      validates :id, :amendable_gid, :emendation_gid, presence: true
+      validates :id, :amendable_gid, :title, :body, :emendation_gid, presence: true
```
`amendable_fields` (in this case `:title` and `:body`) should be dynamic as amendable is configurable.

https://github.com/CodiTramuntana/decidim/blob/feature/amendments-proposals-on-proposals/decidim-core/lib/decidim/amendable.rb#L26

agustibr committed:
Fix proposals/emendations state (#17)

* Fix proposals/emendations state

* Test ProposalMCell proposal/emendation state

* Fix rubocop offences (forgot)

* Refactor amend proposal system spec

* Start fixing admin dashboard

* Fix admin dashboard

* Start fixing amendment form validations

* Refactor amendment form validations

Basic functionality working with redirect_to amendable_gid

* Refactor amendment form validation redirect_to

* Feature/amendments proposals on proposals withdraw (#18)

* Fix emendation withdraw

* Refactor change_proposal_state_to_withdrawn

* Fix proposals_show withdrawn translation missing

* Fix states of proposals

* Button Answer disabled if amendment

* Refactor withdraw to work with "See all withdrawn proposals"

* Check for amendments even if user is not logged in

* Amend Proposal system test

* Fix error when merging cristian/aitor branches

* Check for amendments even if user is not logged in - fixed

* Fix rubocop offences

* Fix mistake when refactoring emendation withdraw

Forgot to change withdraw_proposal.rb back to its original state.

* Add withdraw amendment test

* Fix admin dashboard

* Add red color to withdrawn state in admin manager

* Change proposal_state to resource_state

* Resolve conversations

* Feature/amendments proposals on proposals unable button (#19)

* Unable "Answer" button when the proposal is an emendation

* Fixed offenses with Rubocop

* fixed translate

Commit:
fix rubocop offenses

marc-va committed:
Dynamic validates accepted amendments and test

aitorlb committed:
Fix accept/reject button showing to coauthors

aitorlb committed:
Fix allowed_to_react_to_emendation

aitorlb committed:
Merge branch 'feature/amendments-proposals-on-proposals-accept' of github.com:CodiTramuntana/decidim into feature/amendments-proposals-on-proposals-accept

Commit:
fix translations

Commit:
fix rubocop

marc-va committed:
fix validate accept form
